src/mono/mono/metadata/debug-mono-ppdb.c

```diff
@@ -331,12 +332,15 @@ get_docinfo (MonoPPDBFile *ppdb, MonoImage *image, int docidx)
 	res->hash = (guint8*)mono_metadata_blob_heap (image, cols [MONO_DOCUMENT_HASH]);
 
 	mono_debugger_lock ();
-	cached = (MonoDebugSourceInfo *)g_hash_table_lookup (ppdb->doc_hash, GUINT_TO_POINTER (docidx));
-	if (!cached) {
-		g_hash_table_insert (ppdb->doc_hash, GUINT_TO_POINTER (docidx), res);
-	} else {
-		doc_free (res);
-		res = cached;
+	if (ppdb)
+	{
+		cached = (MonoDebugSourceInfo *)g_hash_table_lookup (ppdb->doc_hash, GUINT_TO_POINTER (docidx));
+		if (!cached) {
+			g_hash_table_insert (ppdb->doc_hash, GUINT_TO_POINTER (docidx), res);
+		} else {
+			doc_free (res);
+			res = cached;
+		}
```

Comment:
if `ppdb == NULL`, are we leaking `res`?  Who is supposed to `free` it?

Maybe we need to add a delta PDB doc cache?